src/supervisor/version.py

```diff
@@ -2,4 +2,5 @@
 This must only include a single line: __version__ = 'x.y.z'
 """
 
+#: This will be automatically set by the publish action when it is uploaded to pypi; only change this for local dev
 __version__ = '3.0.2'
```

Comment:
**todo**: Remove this file and use the `setup.py` property. 
Set the setup.py value to `0.0.0` since CI is managing the version number.

The version value has no purpose during development and is confusing as a static `3.0.2`.

Reply:
If memory serves, we discussed this a while ago. Putting the version in `setup.py` makes it difficult to get it from within the module itself. This is one of the suggested work-arounds.

As for setting to `0.0.0`, I get that it's confusing. I'm wondering if that could cause problems when you've got another package that depends on the current package and you want to see how your changes work. You `pip install -e .[tests]` the current package and then `pip install` the other package that has `current_package >= 1.2.*` in its setup.py. Because the local package is 0.0.0, pip downloads the live version instead of just using the locally-installed development version.

Maybe the solution is set to 0.0.0 but add a note saying to override for local development when necessary?

Reply:
I didn't notice the version being used anywhere other than in the setup.py property. Did my github search fail me?

The scenario you are describing would be the auditor to supervisor relationship? Where auditor depends on supervisor 1.2.* and you have 0.0.0 locally. 

In other ecosystems this would be called package linking. I'm not sure if or how python handles this. I'm assuming you'd have to uninstall supervisor and install into your env from your local directory after doing the parent pip install.

@stdavis are you aware of local package linking in pip?

Reply:
No, I've always just `pip install -e ...` and trusted that I'm getting the local version.

Reply:
That would work for the parent but how about for a dependency of `...` I'm assuming you don't expect the -e to somehow find local requirements.

Reply:
Yes, I do. So if I'm in the `roadkill-skid` conda environment, I can browse to the supervisor repo on my machine and `pip install -e .` and trust that I'm getting the local `supervisor` dependency.

Reply:
right right ok i get ya.

Reply:
Those look like two decent solutions that allow for the 0.0.0 to be in the repo but maybe not persist when developing locally? Does that mean that everyone is on board with this request?

Reply:
>re-install the dependency after installing the other package

This was what I was doing...